spring-cloud-gcp-bigquery/src/main/java/com/google/cloud/spring/bigquery/core/BigQueryTemplate.java

```diff
@@ -247,36 +255,33 @@ public Table createTable(
   public CompletableFuture<WriteApiResponse> writeJsonStream(
       String tableName, InputStream jsonInputStream) {
 
-    CompletableFuture<WriteApiResponse> writeApiFutureResponse =
-        new CompletableFuture<>();
+    CompletableFuture<WriteApiResponse> writeApiFutureResponse = new CompletableFuture<>();
+    Runnable asyncTask =
+        () -> {
+          try {
+            WriteApiResponse apiResponse = getWriteApiResponse(tableName, jsonInputStream);
+            writeApiFutureResponse.complete(apiResponse);
+          } catch (DescriptorValidationException | IOException e) {
+            writeApiFutureResponse.completeExceptionally(e);
+            Thread.currentThread().interrupt();
+            logger.warn(String.format("Error: %s %n", e.getMessage()), e);
+          } catch (Exception e) {
+            writeApiFutureResponse.completeExceptionally(e);
+            // Restore interrupted state in case of an InterruptedException
+            Thread.currentThread().interrupt();
+            logger.warn(String.format("Error: %s %n", e.getMessage()), e);
+          }
+        };
 
-    Thread asyncTask =
-        new Thread(
-            () -> {
-              try {
-                WriteApiResponse apiResponse = getWriteApiResponse(tableName, jsonInputStream);
-                writeApiFutureResponse.complete(apiResponse);
-              } catch (DescriptorValidationException | IOException e) {
-                writeApiFutureResponse.completeExceptionally(e);
-                logger.warn(String.format("Error: %s %n", e.getMessage()), e);
-              } catch (Exception e) {
-                writeApiFutureResponse.completeExceptionally(e);
-                // Restore interrupted state in case of an InterruptedException
-                Thread.currentThread().interrupt();
-              }
-            });
-    asyncTask
-        .start(); // run the thread async so that we can return the writeApiFutureResponse. This
+    executorService.schedule(asyncTask, 0, TimeUnit.MICROSECONDS); // schedule the task with 0 delay
```

Comment:
You might consider the slightly simpler syntax:
```java
CompletableFuture<WriteApiResponse> futureResponse = 
    CompletableFuture.supplyAsync(() -> { ... }, executorService);
```

Reply:
Wired `ExecutorService` and replaced this with `ExecutorService.submit(...)`